File: hooks/tests/fixtures/codenav/zig/navigation.zig
const std = @import("std");
const Counts = struct {
 value: u32,
 pub fn total(self: Counts) u32 { return normalize(self.value); }
};
fn normalize(x: u32) u32 { return x + 1; }
pub fn main() void { const x = normalize(1); _ = x; }

File: hooks/tests/fixtures/codenav/haskell/navigation.hs
module Reads where
import Data.List (sort)
data Sample = Sample Int
normalize :: Int -> Int
normalize x = abs x
total :: [Int] -> Int
total xs = sum (map normalize (sort xs))
class Countable a where
 count :: a -> Int
instance Countable Sample where
 count (Sample n) = n
